internals/cli/cmd_help.go

```diff
@@ -103,17 +105,19 @@ func (cmd *cmdHelp) setParser(parser *flags.Parser) {
 // - duplicated TP lines that break older groff (e.g. 14.04), lp:1814767
 type manfixer struct {
 	bytes.Buffer
-	done bool
+	done        bool
+	programName string
 }
 
 func (w *manfixer) Write(buf []byte) (int, error) {
 	if !w.done {
 		w.done = true
-		if bytes.HasPrefix(buf, []byte(".TH pebble 1 ")) {
+		prefix := ".TH " + w.programName + " "
```

Comment:
We probably had this discussion already ages ago, but can you please explain again why this changes from `".TH pebble 1 "` to `".TH pebble "`, and why the slicing lengths change below?

Reply:
The fact that I don't recall the details about this issue validates your point even further (: I'll look at it, fail again to find a better way to do this, and really document it this time.

(Also: see https://github.com/canonical/go-flags/issues/5)

Reply:
So, apparently this is not changing from `.TH pebble 1 ` to `.TH pebble `. What it's doing is identify the `.TH <program name> ` prefix, find it, and replace the man section number afterwards by an `8` (instead of the default `1`).

The slicing lengths change from fixed magic numbers to the actual lengths of the prefix so that the code appears to be less magic than it was before.

The `w.programName` bit comes from the actual go-flags understanding of what is the program name (`cmd.parser.Name`).

Reply:
Thanks. If you could add an updated comment about what's going on here, that would be great.